src/pharmpy/tools/run_amd.py

```diff
@@ -216,25 +208,30 @@ def _run_resmod(model, path):
 
 
 def _run_covariates(model, continuous, categorical, path):
-    parameters = ['CL', 'VC', 'KMM', 'CLMM', 'MDT', 'MAT', 'QP1', 'QP2', 'VP1', 'VP2']
-
     if continuous is None:
-        covariates = categorical
-    elif categorical is None:
-        covariates = continuous
-    else:
-        covariates = continuous + categorical
+        continuous = []
+        for col in model.datainfo:
+            if col.type == 'covariate' and col.continuous is True:
+                continuous.append(col.name)
+    con_covariates = [Symbol(item) for item in continuous]
 
-    relations = dict()
-    for p in parameters:
-        if model.statements.find_assignment(p):
-            expr = model.statements.before_odes.full_expression(p)
-            for eta in model.random_variables.etas:
-                if eta.symbol in expr.free_symbols:
-                    relations[p] = covariates
-                    break
-    res = run_scm(model, relations, continuous=continuous, categorical=categorical, path=path)
-    return res.final_model
+    if categorical is None:
+        categorical = []
+        for col in model.datainfo:
+            if col.type == 'covariate' and col.continuous is False:
+                categorical.append(col.name)
+    cat_covariates = [Symbol(item) for item in categorical]
+
+    if continuous is not None or categorical is not None:
+        covariates_search_space = (
+            f'CONTINUOUS({con_covariates}); CATEGORICAL({cat_covariates})\n'
+            f'COVARIATE(@IIV, @CONTINUOUS, exp, *)\n'
+            f'COVARIATE(@IIV, @CATEGORICAL, cat, *)'
```

Comment:
You can also refer to all PK parameters with `*` instead of `@IIV`. Not sure which one is better @rikardn?

Reply:
Previously, SCM use this IIV strategy, maybe this is our starting point?

Reply:
Perhaps for now `*` could mean the same as `@IIV`, but if we find new ways of detecting individual parameters we can enhance the `*` option.